config/src/vespa/config/frt/frtconnection.cpp

```diff
@@ -78,41 +81,29 @@ FRTConnection::setError(int errorCode)
 
 void FRTConnection::setSuccess()
 {
+    std::lock_guard guard(_lock);
     _transientFailures = 0;
     _fatalFailures = 0;
-    _suspendedUntil = system_time();
+    _suspendedUntil = steady_time();
 }
 
 void FRTConnection::calculateSuspension(ErrorType type)
 {
     duration delay = duration::zero();
+    steady_time now = steady_clock::now();
+    std::lock_guard guard(_lock);
     switch(type) {
     case TRANSIENT:
-        _transientFailures.fetch_add(1);
-        delay = _transientFailures.load(std::memory_order_relaxed) * getTransientDelay();
-        if (delay > getMaxTransientDelay()) {
-            delay = getMaxTransientDelay();
-        }
+        delay = std::min(6u, ++_transientFailures) * _transientDelay;
         LOG(warning, "Connection to %s failed or timed out", _address.c_str());
         break;
     case FATAL:
-        _fatalFailures.fetch_add(1);
-        delay = _fatalFailures.load(std::memory_order_relaxed) * getFatalDelay();
-        if (delay > getMaxFatalDelay()) {
-            delay = getMaxFatalDelay();
-        }
+        delay = std::min(6u, ++_fatalFailures) * _fatalDelay;
```

Comment:
What is the magic constant 6? Consider giving it a name.

Reply:
MAX_DELAY_MULTIPLIER